executor/index_lookup_hash_join.go

```diff
@@ -215,9 +297,20 @@ func (ow *indexHashJoinOuterWorker) buildTask(ctx context.Context) (*indexHashJo
 	if task == nil || err != nil {
 		return nil, err
 	}
+	var (
+		resultCh            chan *indexHashJoinResult
+		matchedInnerRowPtrs [][]chunk.RowPtr
+	)
+	if ow.keepOuterOrder {
+		resultCh = make(chan *indexHashJoinResult, cap(ow.taskCh))
```

Comment:
We should use a fixed constant here, or the number of concurrency * concurrency in-flight tasks will running.

Reply:
1. No, there will only be `concurrency` tasks at the same time. 
2. `concurrency * concurrency` join result will be produced at the start of the execution.
3. `concurrency` outer result and inner result will be buffered in memory at the same time.